lib/srv/app/server.go

```diff
@@ -1139,6 +1144,19 @@ func (s *Server) getProxyPort() string {
 	return port
 }
 
+// sessionStartTime fetches the session start time based on the the certificate
+// valid date.
+func (s *Server) sessionStartTime(ctx context.Context) time.Time {
+	var startTime time.Time
+	if userCert, err := authz.UserCertificateFromContext(ctx); err == nil {
```

Comment:
I assume we use the certificate NotBefore because we want this to be deterministic for a user, correct?

It would be nice if we could just draw a random number - it's simple, less prone to time clashes (although with nanos precision that is quite difficult) and we could use that to guarantee that this number always exists, dropping the old logic.

Reply:
Indeed interesting approach. Chunks are every five minutes if I recall. So using nanoseconds ensures chunks are relatively in order. However in HA case, time on different app servers may not be perfectly in sync.

Reply:
> I assume we use the certificate NotBefore because we want this to be deterministic for a user, correct?

The idea is to keep the index characteristics as an always-growing number. Since the app sessions are stateless, I relied on the certificate `NotBefore` time since it is consistent during the entire session.

> It would be nice if we could just draw a random number

The random numbers would work here because our events storage doesn't rely on the index to sort (they use the event timestamps). However, it is not guaranteed that the index will always increase within the session. If we're okay with changing this index characteristic, I don't see any problem using a random number.

> However in HA case, time on different app servers may not be perfectly in sync.

We can still have collisions in that case, although they will be rare (given that we're using nanoseconds elapsed time). The solution for this would be to consider the ServerID while generating the index, which is similar to what Snowflake IDs do, or set the index on the proxy/auth servers.